docs/content/drafts/DropdownMenu2.mdx

```diff
@@ -0,0 +1,308 @@
+                >
+                  <type.icon /> {type.name}
+                </ActionList.Item>
+              ))}
+            </ActionList>
+          </DropdownMenu.Overlay>
+        </DropdownMenu>
+      </Box>
+    )
+  }}
+</State>
+
+<br />
+
+```js
+import {DropdownMenu} from '@primer/react/drafts'
+```
+
+<br />
+
+## Examples
+
+### Minimal example
+
+`DropdownMenu` ships with `DropdownMenu.Button` which is an accessible trigger for the overlay. It's recommended to compose `ActionList` with `DropdownMenu.Overlay`
+
+&nbsp;
+
+```javascript live noinline
+// import {DropdownMenu, ActionList} from '@primer/react/drafts'
+const {DropdownMenu, ActionList} = drafts // ignore docs silliness; import like that ↑
+
+const fieldTypes = [
+  {icon: TypographyIcon, name: 'Text'},
+  {icon: NumberIcon, name: 'Number'},
+  {icon: CalendarIcon, name: 'Date'},
+  {icon: SingleSelectIcon, name: 'Single select'},
+  {icon: IterationsIcon, name: 'Iteration'}
+]
+
+const Example = () => {
+  const [selectedIndex, setSelectedIndex] = React.useState(1)
+  const selectedType = fieldTypes[selectedIndex]
+
+  return (
+    <DropdownMenu>
+      <DropdownMenu.Button aria-label="Select field type" leadingIcon={selectedType.icon}>
+        {selectedType.name}
+      </DropdownMenu.Button>
+      <DropdownMenu.Overlay width="medium">
+        <ActionList>
+          {fieldTypes.map((type, index) => (
+            <ActionList.Item key={index} selected={index === selectedIndex} onSelect={() => setSelectedIndex(index)}>
+              <type.icon /> {type.name}
+            </ActionList.Item>
+          ))}
+        </ActionList>
+      </DropdownMenu.Overlay>
+    </DropdownMenu>
+  )
+}
+
+render(<Example />)
+```
+
+### Customise Button
+
+`Dropdown.Button` uses `Button v2` so you can pass props like `variant` and `leadingIcon` that `Button v2` accepts.
+
+```javascript live noinline
+// import {DropdownMenu, ActionList} from '@primer/react/drafts'
+const {DropdownMenu, ActionList} = drafts // ignore docs silliness; import like that ↑
+
+const Example = () => {
+  const [duration, setDuration] = React.useState(1)
+
+  return (
+    <DropdownMenu>
+      <DropdownMenu.Button variant="invisible" aria-label="Select iteration duration">
+        {duration} {duration > 1 ? 'weeks' : 'week'}
+      </DropdownMenu.Button>
+      <DropdownMenu.Overlay width="medium">
+        <ActionList>
+          {[1, 2, 3, 4, 5, 6].map(weeks => (
+            <ActionList.Item key={weeks} selected={duration === weeks} onSelect={() => setDuration(weeks)}>
+              {weeks} {weeks > 1 ? 'weeks' : 'week'}
+            </ActionList.Item>
+          ))}
+        </ActionList>
+      </DropdownMenu.Overlay>
+    </DropdownMenu>
+  )
+}
+
+render(<Example />)
+```
+
+### With External Anchor
+
+To create an anchor outside of the menu, you need to switch to controlled mode for the menu and pass it as `anchorRef` to `DropdownMenu`:
```

Comment:
Could we elaborate on this? And should it be uncontrolled mode instead? Or maybe neither as it appears to run in controlled mode by default? and we're not really mutating ref state either?

Reply:
> And should it be uncontrolled mode instead?

When you pull the open/close out, that's controlled mode. By default, you don't have to control that. Changed the text a little to make that clear.